packages/loader/driver-utils/src/storageUtils.ts

```diff
@@ -0,0 +1,39 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { IPartialSnapshotWithContents } from "@fluidframework/driver-definitions";
+import { ISnapshotTree } from "@fluidframework/protocol-definitions";
+
+/**
+ * Utility API to check if the type of snapshot contents is `IPartialSnapshotWithContents`.
+ * @internal
+ * @param obj - obj whose type needs to be identified.
+ */
+export function isPartialSnapshot(obj: ISnapshotTree): obj is IPartialSnapshotWithContents {
+	return (
+		typeof obj === "object" &&
+		obj !== null &&
+		"partialSnapshot" in obj &&
+		obj.partialSnapshot === true
+	);
+}
+
+/**
+ * Api to extract the contents which belongs to ISnapshotTree from IPartialSnapshotWithContents.
+ * @internal
+ * @param snapshot - snapshot from which ISnapshotTree needs to be extracted.
+ * @returns - instance of ISnapshotTree
+ */
+export function extractISnapshotTreeFromPartialSnapshot(snapshot: ISnapshotTree): ISnapshotTree {
+	if (isPartialSnapshot(snapshot)) {
+		return {
+			id: snapshot.id,
+			blobs: snapshot.blobs,
+			trees: snapshot.trees,
+			unreferenced: snapshot.unreferenced,
```

Comment:
You can as well do nothing and return snapshot. Sure, it will have more stuff on it, but that's fine - interface type does not assume that it can't be extended - any extension is totally valid.

Reply:
We need this api as if you see the usage, we can not supply all of it because like in pendingLocalState, where they Json stirngify the whole snapshot, we can't have arraybuffer in the snapshot which is present in IPartialsnapshot.